eng/pipelines/templates/steps/build-test.yml

```
@@ -86,7 +86,7 @@ steps:
     inputs:
       targetType: 'filePath'
       filePath: ./eng/scripts/Build_Perf.ps1
-      arguments: '${{ parameters.ServiceDirectory }}'
+      arguments: '${{ parameters.ServiceDirectory }} ${{ variables['Use.AzcoreFromMain'] }}'
```

The parameters vs. variables syntax is annoying. You have to do it this way to delay resolution of the symbol, otherwise it will try to calculate the variable before it is set when it renders the yaml to pass to the worker agent. You also need to add an extra `$` so it renders as valid powershell (`$true` vs. `true`).

```suggestion
      arguments: '${{ parameters.ServiceDirectory }} $$(Use.AzcoreFromMain)'
```

However, you also will then need to default it to false [here](https://github.com/Azure/azure-sdk-for-go/blob/main/eng/pipelines/templates/variables/globals.yml). Otherwise devops will fail to find a variable and will pass the literal `$(<var>)` which will fail powershell execution due to syntax...

Yeah devops is special...we typically try to limit use of variables for this reason, but we have a good reason to add it here. The other approach you could consider is just checking for the existence of an env variable in the perf script instead, but since the variable has a `.` you have to do more fancy things to look it up (or remove the `.` from the var name)...

---

Thanks, I've made the change.

I like that the perf script takes it as a parameter so it's somewhat clear this is an option.

---

Should I remove the default value from the build_perf.ps1?

---

I would keep it there so it can be run in other contexts without the extra param, if needed.